diff --git a/drivers/hid/ipts/desc.h b/drivers/hid/ipts/desc.h
index 307438c7c..4d75ce893 100644
--- a/drivers/hid/ipts/desc.h
+++ b/drivers/hid/ipts/desc.h
@@ -14,7 +14,7 @@
 #define IPTS_HID_REPORT_DATA	    65
 #define IPTS_HID_REPORT_SET_MODE    66
 
-#define IPTS_HID_REPORT_DATA_SIZE 7485
+#define IPTS_HID_REPORT_DATA_SIZE 3072
 
 /*
  * HID descriptor for singletouch data.
@@ -66,7 +66,7 @@ static const u8 ipts_fallback_descriptor[] = {
 	0x81, 0x02,	  /*      Input (Variable),              */
 	0x09, 0x61,	  /*      Usage (Gesture Char Quality),  */
 	0x75, 0x08,	  /*      Report Size (8),               */
-	0x96, 0x3D, 0x1D, /*      Report Count (7485),           */
+	0x96, 0xF6, 0x0B, /*      Report Count (7485),           */
 	0x81, 0x03,	  /*      Input (Constant, Variable),    */
 	0x85, 0x42,	  /*      Report ID (66),                */
 	0x06, 0x00, 0xFF, /*      Usage Page (FF00h),            */
diff --git a/drivers/hid/ipts/hid.c b/drivers/hid/ipts/hid.c
index e34a1a4f9..c241782d5 100644
--- a/drivers/hid/ipts/hid.c
+++ b/drivers/hid/ipts/hid.c
@@ -139,7 +139,6 @@ int ipts_hid_input_data(struct ipts_context *ipts, u32 buffer)
 
 	if (header->size == 0)
 		return 0;
-
 	if (header->type == IPTS_DATA_TYPE_HID)
 		return hid_input_report(ipts->hid, HID_INPUT_REPORT, header->data, header->size, 1);
 
